Mooin 405 hide frontpage elements

muschthom opened the pull request:
Es gibt ein globales Menü, hier können folgende globale Änderugen vorgenommen werden unter /admin/settings.php?section=formatsettingmooin4: 
- Links im Kursindex anzeigen
- mooin-Kacheln global verfügbar (Kursstartseite) -> wenn Option nicht gesetzt ist, ist auch in den Kurseinstellungen diese Option nicht verfügbar 

Folgende Elemente können lokal für jeden Kurs ausgeblendet werden (Kurssettings): 
- Kursindex
- News-Bereich
- Progress Bar
- Badge- und Zertifikate-Kachel (wenn globale Settings das erlauben, ansonsten nicht sichtbar)
- Diskussions- und TN-Kachel (wenn globale Settings das erlauben, ansonsten nicht sichtbar)

Tests: 

- [x] mooin_405-Theme aktualisieren
- [x] verschiedene Optionen im globalen Settings unter /admin/settings.php?section=formatsettingmooin4 ein- und ausschalten und schauen, ob im Kurs alles weiterhin gut aussieht und das entsprechende Element ausgeblendet ist
- [ ] in den Kurssettings schauen (entsprechend den globalen Settings), ob die Funktionalität vorhanden/nicht vorhanden ist 
- [x] Optionen in den Kurssettings ein- und ausschalten und schauen, ob im Kurs alles weiterhin gut aussieht und das entsprechende Element ausgeblendet ist

Commit:
Move options to course settings

Commit:
Toggle news section visibility + courseindex link

Commit:
Toggle progress bar visibility

Commit:
Toggle discussion preview tile visibility and index link

Commit:
Toggle mobile discussion button visibility

Commit:
Restore global settings

Commit:
Toggle badge tile visibility + index link

Commit:
Toggle badge & certificate tile + index link

Commit:
Refactor index link visibility

Commit:
Toggle community tile

Commit:
Hide coursefrontpage__side to align main content

If no tiles in coursefrontpage__side, justify main content to center, see also changes in mooin4 theme

Commit:
Fixed mobile view visibility

Commit:
Clear debugging-messages

Commit:
Clear debugging messages

Commit:
undo mustache changes

Commit:
Toggle course index visibility

Commit:
Global settings for tile visibility

Course settings options only appear when globally activated

Commit:
no version update

tinjohn commented:
Im Moment gibt es noch einen Fehler, wenn man über das Frontend zu http://local.onlinecampus-pflege.de/moodle405/admin/settings.php?section=theme_mooin4 geht. Ich prüfe das selbst. - Ist ja theme also ein anderes Issue auf anderem Repo.

tinjohn commented:
Wenn die Kacheln ausgeblendet werden, dann blendet sich auch der entsprechende couse index link aus. Soll das so sein?  Also ich könnte also nur Link und Kachel oder Kacheln ohne Link aber nicht Link OHNE Kacheln?

Commit:
Link option only if tile + link available

Changed lang strings to make functionality more precise;
add js-function to disable link option, if global tile+link option is not available

Commit:
Update lang string

tinjohn reviewed (approved):
funktioniert und ist verständlich, so wie du es abgedacht hattest
